### 1. Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.


In [1]:
import numpy as np
from scipy import stats

In [4]:
y1 = np.array([173, 175, 180, 178, 177, 185, 183, 182])
y2 = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
y3 = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])
n = len(y1) + len(y2) + len(y3)
n

28

In [3]:
stats.f_oneway(y1,y2,y3)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)

Значение $F_{крит}$ можно найти в таблице критических точек распределения Фишера-Снедекора для заданного уровня значимости $\alpha = 0.05$ и двух степеней свободы:

$df_{v1}$ = k - 1 = 3 - 1 = 2 и $df_{v2}$ = n - k = 28 - 3 = 25.

$F_{крит} = 3.385$

А также можно расчитать:


In [5]:
k = 3
alpha = 0.05
F_crit = stats.f.ppf(1 - alpha, (k - 1), (n - k))
F_crit

3.3851899614491687

Так как $F_H$ = 5.5 > $F_{крит}$ = 3.4

различие среднего роста взрослых футболистов, хоккеистов и штангистов является статистически значимым.